### ISSUE TITLE
xdrip broadcasting issue to aaps/nightwatch

### ISSUE THREAD

**Issue by soko78** [NightscoutFoundation/xDrip#419]
**I think this is the same issue as in #401 **
I do not use a transmitter for my libre. I scanned two or three times my libre in the night and aaps did get the data from xdrip but somehow flatlines them. The data are uploaded to Nightscout by xdrip but aaps does not download them from NS. Aaps gets the data by broadcast directly from xdrip.

Then I installed nightwatch (to see if there are issues with broadcasting by xdrip). On the first pic you can see xdrip(nightly from 30.4.18), nightwatch and aaps rc2. In the morning at 8a.m. I installed an older version of xdrip (nightly from 03.03.18) because I thought the problem is because xdrip has a bug. But in both pics you can see that either nightwatch and aaps get somehow "wrong" data or interpret them wrong.

![xdrip vs xdrip](https://user-images.githubusercontent.com/34250545/39614479-7bdd983c-4f70-11e8-89eb-12fc8262d7f9.jpg)

**Comment by dideldum** [comment]
I had a similar behaviour in xDrip+ (2018-03-15) when I switched the Blucon receiver device and scanned with the new device via NFC.

I was using a full-Android-watch (Finow M9) as a normal master receiver to a Blucon on the Libre. As the watch was not worn all the time, there were some gaps. So I made the watch being a follower, configured a (follower) smartphone with NFC to be the receiver/master and scanned the Libre with NFC. The new master got the current value in xDrip+ but instead of filling the gaps with real values, it created a flat line over the last 8 hours with the current value.

This happened only in xDrip+, broadcasting was not involved.
